tests/executables_src/testStatusMonitor.cc

```diff
@@ -0,0 +1,961 @@
+
+  ctk::TestFacility test;
+  test.runApplication();
+  // app.dumpConnections();
+
+  auto requiredValue = test.getScalar<int64_t>(std::string("/parameters/requiredValue"));
+  requiredValue = 409;
+  requiredValue.write();
+  test.stepApplication();
+
+  auto watch = test.getScalar<int64_t>(std::string("/input/path"));
+  watch = 409;
+  watch.write();
+  test.stepApplication();
+
+  auto status = test.getScalar<int32_t>(std::string("/output/path"));
+  status.readLatest();
+
+  // should be in OK state.
+  BOOST_CHECK(status == static_cast<int>(ChimeraTK::StatusOutput::Status::OK));
+
+  // drop in a disable test.
+  auto disable = test.getScalar<ChimeraTK::Boolean>("/parameters/disable");
+  disable = 1;
+  disable.write();
+  test.stepApplication();
+  status.readLatest();
+  BOOST_CHECK(status == static_cast<int>(ChimeraTK::StatusOutput::Status::OFF));
+
+  disable = 0;
+  disable.write();
+  test.stepApplication();
+  status.readLatest();
+  BOOST_CHECK(status == static_cast<int>(ChimeraTK::StatusOutput::Status::OK));
+
+  // set watch value different than required value
+  watch = 414;
+  watch.write();
+  test.stepApplication();
+  status.readLatest();
+  // should be in FAULT state.
+  BOOST_CHECK(status == static_cast<int>(ChimeraTK::StatusOutput::Status::FAULT));
+
+  // drop in a disable test.
+  disable = 1;
+  disable.write();
+  test.stepApplication();
+  status.readLatest();
+  BOOST_CHECK(status == static_cast<int>(ChimeraTK::StatusOutput::Status::OFF));
+
+  disable = 0;
+  disable.write();
+  test.stepApplication();
+  status.readLatest();
+  BOOST_CHECK(status == static_cast<int>(ChimeraTK::StatusOutput::Status::FAULT));
+
+  watch = 409;
+  watch.write();
+  test.stepApplication();
+  status.readLatest();
+  // should be in OK state.
+  BOOST_CHECK(status == static_cast<int>(ChimeraTK::StatusOutput::Status::OK));
+
+  // set requiredValue value different than watch value
+  requiredValue = 413;
+  requiredValue.write();
+  test.stepApplication();
+  status.readLatest();
+  // should be in WARNING state.
+  BOOST_CHECK(status == static_cast<int>(ChimeraTK::StatusOutput::Status::FAULT));
+
+  // set requiredValue value equals to watch value
+  requiredValue = 409;
+  requiredValue.write();
+  test.stepApplication();
+  status.readLatest();
+  // should be in WARNING state.
+  BOOST_CHECK(status == static_cast<int>(ChimeraTK::StatusOutput::Status::OK));
+
+  // check that the tags are applied correctly
+  BOOST_CHECK(status == test.readScalar<int32_t>("/MonitorOutput/output/path"));
+  BOOST_CHECK(requiredValue == test.readScalar<int64_t>("/MonitorParameters/parameters/requiredValue"));
+  disable = 1;
+  disable.write();
+  test.stepApplication();
+  BOOST_CHECK(disable == test.readScalar<ChimeraTK::Boolean>("/MonitorParameters/parameters/disable"));
+}
+
+/*********************************************************************************************************************/
```

Comment:
General comment about the entire file: I would suggest to re-order the tests such that the "main" test about the core functionality of each monitor comes first and then the initial value propagation (and data validity propagation). This helps a bit understanding what goes wrong if a test fails. Typically one looks at the first error first, and if this is about initial value propagation but the module always computes the wrong status this might be confusing.

Also the "chrome bars" should again be between all test cases, and the bigger blocks (each monitor type) should be separated by multiple bars.

Reply:
I tried to improve code formating and order according to your remarks.